plugins/scaffolder-backend/src/scaffolder/actions/builtin/github/OctokitProvider.ts

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright 2021 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { InputError } from '@backstage/errors';
+import {
+  GithubCredentialsProvider,
+  ScmIntegrationRegistry,
+} from '@backstage/integration';
+import { Octokit } from '@octokit/rest';
+import { parseRepoUrl } from '../publish/util';
+
+export type OctokitIntegration = {
+  client: Octokit;
+  token: string;
+  owner: string;
+  repo: string;
+};
+/**
+ * OctokitProvider provides Octokit client based on ScmIntegrationsRegistry configuration.
+ * OctokitProvider supports GitHub credentials caching out of the box.
+ */
+export class OctokitProvider {
```

Comment:
this one isn't exported in the index file - should this perhaps now be part of the public contract to make things easier for adopters that want to make custom actions?

Reply:
Yes, I agree we can export it. Initially `helpers.ts` was designed as an internal tool but I agree it will come handy in building custom actions.